### ISSUE TITLE
Mirror with option to avoid rm -r

### ISSUE THREAD

**Issue by eldondevcg** [lavv17/lftp#308]
Currently, if one creates a script with or executes a mirroring with --delete, then the script may result in running `rm -r` one or more times. Often, the same thing as `rm -r` can be accomplished by descending into the directory, removing the files in the directory, and then `rmdir`ing the directory. It would be nice if it were possible to do that (possibly slower, but possibly safer) action in the context of mirroring.

**Comment by lavv17** [comment]
What would be the purpose of this? Keeping excluded files intact?

**Comment by eldondevcg** [comment]
Correct, ensuring that an excluded file (for example a .git or .keep file) is excluded in a deep subdirectory. Also for logging purposes. If for example it would be useful to have a comprehensive record of all files removed, an rm -r might delete any set of files, but perhaps we would like to ensure that the name of every file deleted is logged.

**Comment by eldondevcg** [comment]
I would be interested in some guidance about how to do this if you can suggest some locations in the source where this might be implemented. I looked through some of the source, but haven't gotten very far, pointers would be appreciated.

**Comment by lavv17** [comment]
I think it would be necessary to create a sub-mirror with an option or empty src-directory parameter instead of "rm -r" job. Then treat this option as an empty source and then remove the target directory itself.
